src/clojure/nrepl/cmdline.clj

```diff
@@ -401,13 +407,21 @@ Exit:      Control+D or (exit) or (quit)"
   Returns connection header string."
   [server options]
   (let [transport (:transport options)
-        port (:port server)
-        ^java.net.ServerSocket ssocket (:server-socket server)
-        host (.getHostName (.getInetAddress ssocket))]
+        ^java.net.ServerSocket ssocket (:server-socket server)]
     ;; The format here is important, as some tools (e.g. CIDER) parse the string
     ;; to extract from it the host and the port to connect to
-    (format "nREPL server started on port %d on host %s - %s://%s:%d"
-            port host (transport/uri-scheme transport) host port)))
+    (let [unix-socket (nrepl.server/unix-server-socket-class)]
+      (if (some-> unix-socket (instance? ssocket))
+        ;; Q: str or .toASCIIString?
```

Comment:
First time I hear of `.toASCIIString`. What would be the benefit of using it?

Reply:
I don't recall for sure yet, but I think I was wondering what junixsocket expects/allows here because I believe if you have a URI with non-ASCII characters, then as long as they're not in the `other` category, `str` (i.e. `toString`) will just pass them through (not RFC compliant), while `.toASCIIString` will octet escape them (compliant).  So if junixsocket can handle the non-compliant chars, then great, but if not, we might need the encoding?

Reply:
See also https://docs.oracle.com/javase/8/docs/api/java/net/URI.html (particularly "These operations are exposed in the constructors and methods of this class as follows: ...").

Reply:
Fair enough. I'd add some comment expanding on this then, so at least your thoughts on the subject don't get lost.

Reply:
I imagine what we really want here is test coverage, i.e. do we return a URI that can make a round trip for "other" characters?  If I get a chance, I might see if I can come up with some relevant tests, and please let me know if you think that's critical for this first pass.  I suppose in the end, what we'll *really* care about is what works correctly with the new JDK functions.  Offhand, I'd guess toASCIIString, but don't know yet.

Reply:
I'm with the current state of the code.